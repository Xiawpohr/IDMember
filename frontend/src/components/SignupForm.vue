<template>
  <form @submit.prevent="$emit('submitted', auth)" >
    <input type="text" placeholder="Email" v-model="email">
    <input type="password" placeholder="Password" v-model="password">
    <input type="password" placeholder="Password Again" v-model="confirmedPassword">
    <button type="submit">Sign Up</button>
  </form>
</template>

<script>
export default {
  data() {
    return {
      email: '',
      password: '',
      confirmedPassword: ''
    }
  },
  computed: {
    auth() {
      return {
        email: this.email,
        password: this.password
      }
    }
  }
}
</script>

<style>

</style>
