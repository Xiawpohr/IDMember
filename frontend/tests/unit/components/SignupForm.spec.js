import { shallowMount } from '@vue/test-utils'
import SignupForm from '@/components/SignupForm.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(SignupForm)
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should call submittd event when submitting the form', () => {
  const auth = { email: 'test@example.com', password: '123456' }
  wrapper.setData({
    ...auth
  })
  const button = wrapper.find('button')
  button.trigger('click')
  button.trigger('submit')
  expect(wrapper.emitted('submitted')[0]).toEqual([auth])
})
