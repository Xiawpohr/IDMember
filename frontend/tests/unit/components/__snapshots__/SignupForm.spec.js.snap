// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `<form><input type="text" placeholder="Email"> <input type="password" placeholder="Password"> <input type="password" placeholder="Password Again"> <button type="submit">Sign Up</button></form>`;
